pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -388,11 +413,24 @@ func (i *KumaInjector) sidecarEnvVars(mesh string, podAnnotations map[string]str
 	return result, nil
 }
 
-func (i *KumaInjector) NewVolumeMounts(pod *kube_core.Pod) []kube_core.VolumeMount {
+func (i *KumaInjector) NewVolumeMounts(pod *kube_core.Pod, need_dd_mount bool) []kube_core.VolumeMount {
+	var volumeMounts []kube_core.VolumeMount
 	if tokenVolumeMount := i.FindServiceAccountToken(pod); tokenVolumeMount != nil {
-		return []kube_core.VolumeMount{*tokenVolumeMount}
+		volumeMounts = append(volumeMounts, *tokenVolumeMount)
+	}
+
+	if need_dd_mount {
+		volumeMounts = append(volumeMounts, *i.buildDatadogVolumeMount())
+	}
+
+	return volumeMounts
+}
+
+func (i *KumaInjector) buildDatadogVolumeMount() *kube_core.VolumeMount {
+	return &kube_core.VolumeMount{
+		Name:      "datadog-vol",
+		MountPath: "/datadog-mount",
```

Comment:
Nit: I was going to suggest qualifying the `/datadog-mount` with something kuma-specific to avoid conflicts, but since this is only the sidecar container, we control all the mounts. I'd probably prefer something like`/var/kuma/datadog`, but I don't feel too strongly about it :)